galaxy_importer/collection.py

```diff
@@ -173,12 +232,105 @@ def _load_collection_manifest(self):
         if not os.path.exists(manifest_file):
             raise exc.ManifestNotFound('No manifest found in collection')
 
+        default_logger.debug('manifest_file: %s', manifest_file)
+
         with open(manifest_file, 'r') as f:
             try:
                 data = schema.CollectionArtifactManifest.parse(f.read())
             except ValueError as e:
                 raise exc.ManifestValidationError(str(e))
-            self.metadata = data.collection_info
+
+            default_logger.debug('data: %s', data)
+            default_logger.debug('data.file_manifest_file: %s', data.file_manifest_file)
+            return data
+
+    def _load_collection_file_manifest(self, path_prefix, file_manifest_file):
+        """Load CollectionArtifactFileManifest data from file_manifest_file
+
+        Args:
+            path_prefix (str): Any file path prefix we need to add to file paths in the
+                CollectionArtifactFile artifact_file
+            file_manifest_file (CollectionArtifactFile): object with info (name, path) of the files
+                in the artifact. ie, "FILES.json"
+
+        Raises:
+            ManifestValidationError: If there are any errors loading, parsing,
+                deserializing, or validating the data in file_manifest_file
+
+        Returns:
+            CollectionArtifactFileManifest: The data from file_manifest_file
+
+        """
+        default_logger.debug('file_manifest_file: %s', file_manifest_file)
+
+        check_artifact_file(path_prefix=path_prefix, artifact_file=file_manifest_file)
```

Comment:
any reason this is called twice?, once in `CollectionLoader.load` right before this class method is called

Reply:
Agreed, one of those calls is redundant. 

I'll remove the call in CollectionImport.load() so it is only used in _load_collection_file_manifest().

Unless we want CollectionLoader.load() to call `check_artifact_file(...FILES.json...)` in a separate step from reading/deserializing/parsing/validating the ds in FILES.json. I don't think we need that now unless we start running into failures where it is hard to tell what step failed.

Reply:
updated